### PR TITLE
Gitter Authentication

### PR DESCRIPTION
**Pull request by Lorac**
Issue #94

Around line 97 in GitterAuthenticationHandler.cs, we receive an json array for the user information. There might be a better way to do this.

### CODE REVIEW

File: Owin.Security.Providers/Gitter/GitterAuthenticationOptions.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        {
+            get { return Description.Caption; }
+            set { Description.Caption = value; }
+        }
+
+        /// <summary>
+        ///     Determines the permissions granted from the application to Gitter.
+        ///     Options include: identify, read, post, client, and/or admin
+        /// </summary>
+        public IList<string> Scope { get; set; }
```
Comment (jerriep) [inline Owin.Security.Providers/Gitter/GitterAuthenticationOptions.cs]: Can you please make the setter for Scope private to be in line with all the other ones? This is also to be in line with the ASP.NET 5 providers.

File: Owin.Security.Providers/Gitter/GitterAuthenticationHandler.cs [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+                {
+                    context.Identity.AddClaim(new Claim("urn:gitter:displayName", context.UserDisplayName, XmlSchemaString, Options.AuthenticationType));
+                }
+                if (!string.IsNullOrEmpty(context.UserGV))
+                {
+                    context.Identity.AddClaim(new Claim("urn:gitter:gv", context.UserAvatarUrlMedium, XmlSchemaString, Options.AuthenticationType));
+                }
+                if (!string.IsNullOrEmpty(context.UserUrl))
+                {
+                    context.Identity.AddClaim(new Claim(ClaimTypes.Webpage, context.UserUrl, XmlSchemaString, Options.AuthenticationType));
```
Comment (jerriep) [inline Owin.Security.Providers/Gitter/GitterAuthenticationHandler.cs:126]: You are setting the Webpage claim type four times to four different values. For the avatar urls below rather use "urn:gitter:whatever" like you did above.

File: Owin.Security.Providers/Gitter/GitterAuthenticationHandler.cs [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+                }
+                if (!string.IsNullOrEmpty(context.UserAvatarUrlSmall))
+                {
+                    context.Identity.AddClaim(new Claim(ClaimTypes.Webpage, context.UserAvatarUrlSmall, XmlSchemaString, Options.AuthenticationType));
+                }
+                if (!string.IsNullOrEmpty(context.UserAvatarUrlMedium))
+                {
+                    context.Identity.AddClaim(new Claim(ClaimTypes.Webpage, context.UserAvatarUrlMedium, XmlSchemaString, Options.AuthenticationType));
+                }
+                if (!string.IsNullOrEmpty(context.UserAvatarUrlMedium))
```
Comment (jerriep) [inline Owin.Security.Providers/Gitter/GitterAuthenticationHandler.cs]: You are setting UserAvatarUrlMedium twice? Copy and paste error somewhere? ;)

File: Owin.Security.Providers/Gitter/GitterAuthenticationOptions.cs (hunk 1 above)
Comment (Lorac) [inline Owin.Security.Providers/Gitter/GitterAuthenticationOptions.cs]: Fixed

File: Owin.Security.Providers/Gitter/GitterAuthenticationHandler.cs (hunk 2 above)
Comment (Lorac) [inline Owin.Security.Providers/Gitter/GitterAuthenticationHandler.cs:126]: Fixed

File: Owin.Security.Providers/Gitter/GitterAuthenticationHandler.cs (hunk 3 above)
Comment (Lorac) [inline Owin.Security.Providers/Gitter/GitterAuthenticationHandler.cs]: Fixed

**Comment by jerriep** [comment]
Sweet, thanks for the contribution Maxime :)

**Comment by jerriep** [comment]
@Lorac We are looking for help moving to ASP.NET 5. Please read http://www.jerriepelser.com/blog/calling-contributors-for-oauth-providers

**Pull request merged**